utility/README.md

```diff
@@ -1,679 +1,1309 @@
-}
+Servicer's are incentivized to respond to any valid relay since it is applicable for reward distribution. The session tokens used for rate limiting by the Servicer will come out of the same bucket as described in the [rate limiting algorithm](#315-rate-limiting) regardless of who in the ring signed the request.
+
+```mermaid
+sequenceDiagram
+    actor A as Application
+    participant WS as World State
+    actor G as Gateway
+    actor S as Servicer
+
+    A ->>+ G: StartSession
+    G ->> G: StartSession
+    G ->>- A: AccessToken
+
+    loop Session Duration
+        A ->> +G: Request(AccessToken)
+        G ->> G: Sign Request
+        G ->> +S: SignedRequest
+        S ->> S: Validate Signature<br>& Session Limits
+        S ->> S: Handle Request<br>& Sign Response
+        S ->> -G: SignedResponse
+        G ->> G: ** Gateway specific features ** <br>(altruist, check, challenge, proof, etc...)
+        G ->> -A: Response
+    end
 ```
 
+#### 3.5.6 Registration
 
-A Validator is able to modify their initial staking values including the ServiceURL, OperatorPubKey, and StakeAmount by submitting a StakeMsg while already staked. It is important to note, that a StakeAmount is only able to be modified greater than or equal to the current value. This allows the protocol to not have to track pieces of stake from any Validator and enables an overall simpler implementation. 
+Registration differs from staking in the sense that the pubKey is known but there are no economic benefits/penalties in this stage of the protocol's progression.
 
+The Gateway must register on-chain in order for the Servicer to accept its signature as part of the ring. Future versions of the protocol may include on-chain rewards or penalties for the Gateway, but the current iteration will incentivize Gateways to provide a high quality, highly trusted service through free market economics.
 
-```
-# Validator(Edit)StakeMsg Interface
-type ValidatorStakeMsg interface {
-  GetPublicKey()  PublicKey       # which Validator stake data is being edited?
-  GetStakeAmount() BigInt         # must be greater than or equal to the current value
-  GetServiceURL() ServiceURL      # may be modified
-  GetOperatorPubKey() PublicKey   # may be modified
+When staking, the Gateway must bond a certain amount of POKT to be able to participate in the network. The governance parameter, `StakePerAppDelegation` limits the number of Applications that can delegate to it, and it is the Gateway's responsibility to increase its stake as the number of Applications that trust it grow.
+
+For example, if `StakePerAppDelegation` is 100 POKT and the Gateway has staked 1000 POKT, a transaction by the 11th Application to delegate to it will be rejected until the stake is increased appropriately. However, if `StakePerAppDelegation` is 0 POKT, all Gateways, which are permissionless actors can have an unbounded number of Applications delegate to them.
```

Comment:
If StakePerAppDelegation value changes, do the App stakes get re-distributed per that new value, or are they grandfathered into the old value and only net new delegating applications take the new value?

Reply:
Perfect!